### PR TITLE
add generating auth files for jenkins profisioned rosa

### PR DESCRIPTION
**Pull request by qiliRedHat**
1. https://issues.redhat.com/browse/OCPQE-15925, updated generate_auth_files.sh and README.md.
2. Update Tasks.py to add a universal __get_kubeconfig(self,user).
3. Add a fix to scale_deployment function in Tasks.py  to avoid the Warning for DeploymentConfig in 4.14.
4. Exclude insights operator as it is degraded on ocm staging env as expected

### CODE REVIEW

**Comment by qiliRedHat** [comment]
Test
```
% ./generate_auth_files.sh                        
CLUSTER_ID, OCM_TOKEN and OCM_ENV are provided for ocm-profile-ci Jenkins job provisioned rosa cluster.
Checking ocm cli
Checking rosa cli
I: Logged in as 'svetsa-ocm' on 'https://api.stage.openshift.com'
Creating admin user ....
admin file is created
Waiting for admin user login and generate kubeconfig ....
admin user login successfully and kubeconfig file is created.
Generating 80 IDP users under the htpasswd idp ....
users file is created
The credential files for reliability test are created under folder path_to_auth_files.
```
```
% ls path_to_auth_files 
admin				create_idp_users_rosa.log	kubeconfig			users
```

**Comment by qiliRedHat** [comment]
@svetsa-rh PTAL

**Comment by qiliRedHat** [comment]
Test on classic rosa
```
# ./generate_auth_files.sh 
CLUSTER_ID, OCM_TOKEN and OCM_ENV are provided for ocm-profile-ci Jenkins job provisioned rosa cluster.
Checking ocm cli
Checking rosa cli
I: Logged in as 'svetsa-ocm' on 'https://api.stage.openshift.com'
Creating admin user ....
admin file is created
Waiting for admin user login and generate kubeconfig ....
admin user login successfully and kubeconfig file is created.
Generating 50 IDP users under the htpasswd idp ....
users file is created
The credential files for reliability test are created under folder path_to_auth_files.

# ls path_to_auth_files/
admin  create_idp_users_rosa.log  kubeconfig  users
```

File: reliability-v2/utils/generate_auth_files.sh [hunk 1]
```diff
@@ -2,31 +2,64 @@
...
 ## For Flexy-install cluster: Export AUTH_PATH which includes kubeconfig, users.spec and kubeadmin-password files that
 ## downloaded from the Build Artifacts of the Flexy-install job.
+## For rosa cluster installed by Prow: Export TOKEN, SERVER, and PROJECT(optional)
 ## For cluster installed by other ways: export KUBECONFIG_PATH.
 ################################################
 
 function _usage {
     cat <<END
 Usage: $(basename "${0}")
+If this is a cluster provisioned by Flexy-install Jenkins job, export AUTH_PATH as the path includes kubeconfig, users.spec and kubeadmin-password files downloaded from the Flexy-install job.
```
Comment (svetsa-rh) [inline reliability-v2/utils/generate_auth_files.sh:15]: So, AUTH_PATH is the directory containing all 3 files?

**Comment by svetsa-rh** [comment]
Hi @qiliRedHat I added couple of minor comments, but looks good overall. 
Couple of questions: 
1) If someone were to look at reliability test results, for ex https://redhat-internal.slack.com/archives/C0266JJ4XM5/p1689929537589349, how can they know if the test ran on Self managed/Classic Rosa or Hypershift cluster? 
2) Are we printing any output to reliability logs, like `oc describe cluster` or `rosa describe cluster`? Can you share sample logs if you have them? Do you think this will be helpful?

FYI, to distinguish between Classic Rosa cluster and Hypershift cluster, you could run a command such as 
`rosa describe cluster -c "$CLUSTER_ID" -o json  | jq -r ".hypershift.enabled"`

File: reliability-v2/README.md [hunk 2]
```diff
@@ -79,6 +58,53 @@ If [start.sh](https://github.com/openshift/svt/tree/master/reliability-v2/start.
...
+3. users: a file contains pairs of usernames and passwords. e.g testuser-0:password1,testuser-1:password2,...,testuser-n:passwordn (No space)
+Use either of the options below to generate the 3 files.
+
+### Option 1: Generate auth files automatically with script
+[generate_auth_files.sh](https://github.com/openshift/svt/blob/master/reliability-v2/utils/generate_auth_files.sh) can help you to generate the required files mentioned above. Get the command's help with generate_auth_files.sh -h.
+
+#### Pre-requisite:
+ - [oc cli](https://console.redhat.com/openshift/downloads)
+
+For rosa and rosa hypershift cluster, additional pre-requisite:
```
[the remark this answers is not in the document]
Comment (qiliRedHat) [reply reliability-v2/README.md:74]: Added

File: reliability-v2/README.md [hunk 3]
```diff
@@ -17,40 +17,19 @@ cd svt/reliability-v2
...
+## Generate auth files automatically with script
 [generate_auth_files.sh](https://github.com/openshift/svt/blob/master/reliability-v2/utils/generate_auth_files.sh) can help you to generate the required files mentioned above. Get the command's help with generate_auth_files.sh -h.
 
-Option 2
-If you installed your cluster with [Flexy-install](https://mastern-jenkins-csb-openshift-qe.apps.ocp4.prod.psi.redhat.com/job/ocp-common/job/Flexy-install/), download kubeconfig, users.spec and kubeadmin-password files from the Build Artifacts after the job completed successfully. The user `kubeadmin` has admin priviledge and its password is in kubeadmin-password. users.spec contains username and password pairs of 50 test users. Then compose the above 3 files based on the files you get.
+#### Pre-requisite:
+ - [oc cli](https://console.redhat.com/openshift/downloads)
 
-If you installed your rosa cluster with Prow CI, follow below instruction to get the kubeconfig, admin credential file and users credential files. Then compose the above 3 files based on the files you get.
+For rosa and rosa hypershift cluster, additional pre-requisite:
```
[the remark this answers is not in the document]
Comment (qiliRedHat) [reply reliability-v2/README.md:26]: Added. Thanks for the comment.

**Comment by qiliRedHat** [comment]
> Hi @qiliRedHat I added couple of minor comments, but looks good overall. Couple of questions:
> 
> 1. If someone were to look at reliability test results, for ex https://redhat-internal.slack.com/archives/C0266JJ4XM5/p1689929537589349, how can they know if the test ran on Self managed/Classic Rosa or Hypershift cluster?
> 2. Are we printing any output to reliability logs, like `oc describe cluster` or `rosa describe cluster`? Can you share sample logs if you have them? Do you think this will be helpful?
> 
> FYI, to distinguish between Classic Rosa cluster and Hypershift cluster, you could run a command such as `rosa describe cluster -c "$CLUSTER_ID" -o json | jq -r ".hypershift.enabled"`

What we currently print to log are just api info and server version.
```
Reliability-v2 test has started on cluster: [https://api.sdq-ci-khiua.7elj.s3.devshift.org:443](https://api.sdq-ci-khiua.7elj.s3.devshift.org/)
Server Version: 4.13.4
Kubernetes Version: v1.26.5+7d22122
```
I also thought about this, but some concern/limitation stop me from doing that
1. For a non rosa cluster, we don't require user to install ocm and rosa command line.  
2. Inside the reliability test, we currently does not support com and rosa command (this can be done in the future out of this pr if there is a need). 
I don't think from oc cli we can identify if the cluster is classic rosa or rosa hypershift, a tricky way to identify from the cluster name/api - the cluster installed by the ocm Jenkins job has a special name 'sdq-ci-'. But again no rosa and ocm support in this pr, so I think we can consider this later.

**Pull request merged**